docs/migrate-from-sidecar-to-ambient/migration.adoc

```diff
@@ -0,0 +1,1604 @@
+** <<common-issues, 5.1 Common Issues>>
+** <<rollback-procedures, 5.2 Rollback Procedures>>
+* <<post-migration-optimization, 6. Post-Migration Optimization>>
+** <<observability-considerations, 6.1 Observability Considerations>>
+** <<configure-kiali-for-ambient-mode, 6.2 Configure Kiali for Ambient Mode>>
+* <<additional-resources, 7. Additional Resources>>
+
+---
+
+[[overview-of-sidecar-to-ambient-migration]]
+== 1. Overview of Sidecar to Ambient Migration
+
+[[migration-strategy]]
+=== 1.1 Migration Strategy
+
+The migration from sidecar to ambient mode follows a structured, phased approach to ensure a smooth transition with minimal disruption. Key principles include:
+
+1. **Gradual Transition**: Both sidecar and ambient proxies operate simultaneously during migration
+2. **Service-by-Service**: Migration can occur incrementally, one service or namespace at a time
+3. **Policy Preservation**: All existing security and traffic policies are maintained throughout the process
+4. **Rollback Support**: Each step can be reversed if issues arise
+
+[[key-architectural-changes]]
+=== 1.2 Key Architectural Changes
+
+[cols="1,2,2"]
+|===
+|Component |Sidecar Mode |Ambient Mode
+
+|**L4 Processing**
+|Envoy sidecar per pod
+|ZTunnel node proxy (shared)
+
+|**L7 Processing**
+|Envoy sidecar per pod
+|Waypoint proxy per namespace or service or workload.
+
+|**mTLS Enforcement**
+|Sidecar proxy
+|ZTunnel proxy
+
+|**Policy Enforcement**
+|Sidecar proxy
+|ZTunnel (L4) + Waypoint (L7)
+|===
+
+---
+
+[[prerequisites]]
+== 2. Prerequisites
+
+This migration guide uses example commands that can be adapted to any application. For reference purposes, we use placeholder values like `<NAMESPACE>`, `<SERVICE_NAME>`, and `<GATEWAY_NAME>` that you should replace with your actual application details.
+
+**Related Documentation:**
+- If you want to follow along with a sample application, you can deploy the bookinfo application by following the steps described in link:../addons/addons.adoc#deploy-gateway-and-bookinfo[Deploying Gateway and Bookinfo]
+- For detailed waypoint proxy configuration, see link:../common/istio-ambient-waypoint.adoc[Waypoint Proxy Documentation]
+- For general application deployment guidance, refer to link:../common/install-bookinfo-app.adoc[Installing the Bookinfo Application]
+
+You can use any application that is currently running with sidecar injection enabled.
+
+[[technical-requirements]]
+=== 2.1 Technical Requirements
+
+Before starting migration, ensure your environment meets these requirements:
+
+For Kubernetes:
+
+- **Kubernetes 1.21+**: Minimum version for ambient mode support
+- **Sail Operator v1.27+**: Must be running Sail Operator version 1.27 or later
+- **Istio v1.27+**: Control plane must be v1.24 or later for ambient support
+- **Cluster Admin Access**: Required for namespace labeling and resource creation
+
+For Openshift:
+
+- **Openshift 4.19+**: Minimum version for ambient mode support
```

Comment:
This is not exactly true. Ambient is also supported on the lower OCP version.
On OCP 4.18 and lower, the user needs to install Gateway API CRD's
```
kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null || \
  kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.3.0/standard-install.yaml
```
Can we add here a note about that?